chain/deals/client.go

```diff
@@ -181,7 +182,7 @@ func (c *Client) Start(ctx context.Context, p ClientDealProposal) (cid.Cid, erro
 		smsg, err := c.mpool.MpoolPushMessage(ctx, &types.Message{
 			To:       actors.StorageMarketAddress,
 			From:     p.Client,
-			Value:    types.BigMul(p.PricePerEpoch, types.NewInt(p.Duration)),
+			Value:    types.BigSub(types.BigMul(p.PricePerEpoch, types.NewInt(p.Duration)), clientMarketBalance.Available),
```

Comment:
This can go negative, which will error out.

Reply:
if clientMarketBalance.Available.LessThan(types.BigMul(p.PricePerEpoch, types.NewInt(p.Duration)))

Reply:
Aaah, right.